### PR TITLE
Add basic crictl info config with sandboxImage

### LINKED ISSUE #6816: Information about pause image requested in runtime status
Since kubernetes 1.27.0, `kubeadm` is now looking for information about the pause image configuration of the runtime.

* https://github.com/kubernetes/kubernetes/issues/106893

```
getting status of runtime: failed to template data: template: tmplExecuteRawJSON:1:9: executing \"tmplExecuteRawJSON\" at <.config.sandboxImage>: map has no entry for key \"config\"
```

It was hardcoded, to look in the containerd config. So now all other container runtimes needs to implement that field.

---

The user can do `crictl info -q`, if just wanting the old status.

With the new info, `kubeadm` will warn if cri-o has not been configured:

https://kubernetes.io/docs/setup/production-environment/container-runtimes/#override-pause-image-cri-o

Without the info, there will be the "info" output (above) but no "warn".

### PR DESCRIPTION
**Pull request by afbjorklund**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Adds information about pause image to `crictl info`, as used by kubeadm in 1.27

#### Which issue(s) this PR fixes:

Fixes #6816

#### Does this PR introduce a user-facing change?

```release-note
a new field "config" has been added to the verbose output of crictl info
```

`crictl info`
```json
{
  "status": {
    "conditions": [
      {
        "type": "RuntimeReady",
        "status": true,
        "reason": "",
        "message": ""
      },
      {
        "type": "NetworkReady",
        "status": false,
        "reason": "NetworkPluginNotReady",
        "message": "Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
      }
    ]
  },
  "config": {
    "sandboxImage": "registry.k8s.io/pause:3.9"
  }
}

```

### CODE REVIEW

File: server/runtime_status.go
Comment (saschagrunert) [inline server/runtime_status.go:49]: I'd say we either pass the `s.config` or we make it a method of `*Server`.

File: server/runtime_status.go
Comment (saschagrunert) [inline server/runtime_status.go:55]: I don't think we need the whole config in the error message, but I'm not blocking on that.

**Review by saschagrunert — COMMENTED** [review]
Thank you, just two nits otherwise LGTM.

File: server/runtime_status.go
Comment (afbjorklund) [reply server/runtime_status.go:55]: I was trying to do the same as with the other verbose "info" fields, for containers and images

File: server/runtime_status.go
Comment (afbjorklund) [reply server/runtime_status.go:49]: Maybe just pass the pause image, since it is an internal function anyway ?

Comment (afbjorklund) [reply server/runtime_status.go:49]: It is a bit of a mix:
```go
func (s *Server) createContainerInfo(container *oci.Container) (map[string]string, error)
func createImageInfo(result *pkgstorage.ImageResult) (map[string]string, error)
```

Comment (afbjorklund) [reply server/runtime_status.go:49]:
```go
func (s *Server) createRuntimeInfo() (map[string]string, error)
```

**Comment by afbjorklund** [comment]
I don't think there is any requirement or definition on "config", other than what kubeadm is looking for...

https://github.com/kubernetes/kubernetes/pull/115610/commits/d12b4d4c6772bc7735c3765d79ecc46c69595088

**Review by saschagrunert — APPROVED** [review]
LGTM, ptal @cri-o/cri-o-maintainers 

/hold
because of auto approval. Maintainers feel free to unhold if it looks good to y'all.

**Comment by afbjorklund** [comment]
Maybe a unit test would improve coverage, but I don't really understand the format of it. Seems both tests are the same ?

**Comment by saschagrunert** [comment]
Yeah that's known

/override ci/prow/e2e-gcp-ovn

**Comment by haircommander** [comment]
/unhold
/lgtm

thanks!

**Pull request merged**
